src/api/app/views/webui/users/notifications/_notification.html.haml

```diff
@@ -47,7 +47,7 @@
                 - elsif avatar_object.class.name == 'Project'
                   %span.fa.fa-cubes.text-secondary.rounded-circle.bg-body-secondary.border.simulated-avatar{
                     title: "Project #{avatar_object}" }
-                - else
+                - elsif avatar_object.respond_to?(:email)
```

Comment:
Before the refactoring of the avatar helper we had:

```
              case avatar_object.class.name
              when 'User', 'Group'
                render(AvatarComponent.new(name: avatar_object.name, email: avatar_object.email, size: 23, shape: :circle))
```

 So we rendered the avatar component when object was 'User' or 'Group', in any other case we displayed nothing.  Isn't it better than `avatar_object.respond_to?(:email)`?

Reply:
I double checked that if non of the conditions are true, I mean, it's not a Project, Package, User or Group, then the `<ul>` doesn't take any room in the UI.

Reply:
For the record: Group has a `title` attribute. The reason why it also responds to `name` is because of this alias.

https://github.com/openSUSE/open-build-service/blob/39d47f2dbab539f9af4598c2c0206a7673b542f1/src/api/app/models/group.rb#L39

Reply:
Yeah, I used the `respond_to?(:email)` to avoid typing `class.name == 'User' || class.name == 'Group'` but maybe it's not that clear that way. I will revert it to use the user and group class names as it's more straightforward.